### PR TITLE
[ChatGPT Quick Actions] Add "Transform" and "Preview Transform" commands

### PR DESCRIPTION
**Pull request by JonathanAquino**
## Description

This PR adds two commands to ChatGPT Quick Actions: Transform and Preview Transform. With it, you can enter a prompt to apply various transformations to the selected text, like code formatting ("beautify java code"), translation ("print each english line followed by the french translation below each english line"), and analysis ("which lines have positive sentiment, which have negative, and which are neutral?").

* **_Transform_** replaces the selected text with transformed text based on a prompt that is entered when the command is run; for example, "beautify json" or "beautify sql" or "find duplicate lines"
* **_Preview Transform_** is the same but outputs to a preview window instead of replacing the selected text.

## Screencast

| Before | Transform | After |
| --- | --- | --- |
|image|Here are the duplicate lines f… 2024-09-13 00-30-33(1)|image|

| Before | Transform | After |
| --- | --- | --- |
|image|select 2024-09-13 00-35-52(1)|image|

| Before | Preview Transform | After |
| --- | --- | --- |
|image|image|image|

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/chatgpt-quick-actions/package.json [hunk 1]
```diff
@@ -296,6 +297,106 @@
           ]
         }
       ]
+    },
+    {
```
Comment (JonathanAquino) [inline extensions/chatgpt-quick-actions/package.json:301]: Pretty much the same as the execute action above, but adds a "prompt" argument.

File: extensions/chatgpt-quick-actions/package.json [hunk 2]
```diff
@@ -296,6 +297,106 @@
...
+            },
+            {
+              "title": "GPT-3.5 Turbo",
+              "value": "gpt-3.5-turbo"
+            }
+          ]
+        }
+      ]
+    },
+    {
```
Comment (JonathanAquino) [inline extensions/chatgpt-quick-actions/package.json:351]: Same as the preview action above, but adds a "prompt" argument.

File: extensions/chatgpt-quick-actions/src/transform.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Clipboard, LaunchProps, getPreferenceValues, getSelectedText, showHUD } from "@raycast/api";
```
Comment (JonathanAquino) [inline extensions/chatgpt-quick-actions/src/transform.tsx:1]: Same as https://github.com/raycast/extensions/blob/main/extensions/chatgpt-quick-actions/src/execute.ts but adds a "prompt" argument.

File: extensions/chatgpt-quick-actions/src/preview_transform.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { LaunchProps, getPreferenceValues } from "@raycast/api";
```
Comment (JonathanAquino) [inline extensions/chatgpt-quick-actions/src/preview_transform.tsx:1]: Same as https://github.com/raycast/extensions/blob/main/extensions/chatgpt-quick-actions/src/preview.tsx but adds a "prompt" argument.

File: extensions/chatgpt-quick-actions/package.json [hunk 5]
```diff
@@ -296,6 +297,106 @@
...
+      "name": "preview_transform",
+      "title": "Preview Transform",
```
Comment (one-data-cookie) [inline extensions/chatgpt-quick-actions/package.json:353]: I'd personally consider naming it "Transform Preview" as this is – from UX standpoint – more tied to "Transform" then "Preview".

File: extensions/chatgpt-quick-actions/package.json [hunk 6]
```diff
@@ -296,6 +297,106 @@
...
+      ],
+      "preferences": [
+        {
+          "name": "model_preview_transform",
+          "description": "Override the global preferred model for this command",
+          "type": "dropdown",
+          "required": false,
+          "title": "Model Override",
+          "placeholder": "Follow global model",
+          "data": [
```
Comment (one-data-cookie) [inline extensions/chatgpt-quick-actions/package.json:372]: Seeing `data` repeated verbatim across the code, is it maybe time to define this once and reuse it?

File: extensions/chatgpt-quick-actions/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,10 @@
 # ChatGPT Quick Actions Changelog
 
+## [Add Transform and Preview Transform commands] - 2024-09-12
```
Comment (pernielsentikaer) [inline extensions/chatgpt-quick-actions/CHANGELOG.md:3]:
```suggestion
## [Add Transform and Preview Transform commands] - {PR_MERGE_DATE}
```

**Comment by pernielsentikaer** [comment]
@alanzchen do you also want to check this?

**Comment by alanzchen** [comment]
I have no objections

File: extensions/chatgpt-quick-actions/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,10 @@
...
+
+- Add a Transform command which replaces the selected text with transformed text based on a prompt that is entered when the command is run.
+- Add a Preview Transform command which does the same but displays the result in a preview window instead of replacing the selected text.
```
[the remark this answers is not in the document]
Comment (JonathanAquino) [reply extensions/chatgpt-quick-actions/CHANGELOG.md]: @one-data-cookie I updated the changelog text as requested.

File: extensions/chatgpt-quick-actions/package.json (hunk 5 above)
Comment (JonathanAquino) [reply extensions/chatgpt-quick-actions/package.json]: @one-data-cookie Changed "preview transform" to "transform preview" as requested.

File: extensions/chatgpt-quick-actions/package.json (hunk 6 above)
Comment (JonathanAquino) [reply extensions/chatgpt-quick-actions/package.json:372]: @one-data-cookie I'm not sure how to reuse data blocks in package.json - do you have any tips or can you link me to an example?

File: extensions/chatgpt-quick-actions/CHANGELOG.md (hunk 7 above)
Comment (JonathanAquino) [reply extensions/chatgpt-quick-actions/CHANGELOG.md]: @pernielsentikaer I switched to {PR_MERGE_DATE} as requested. is it documented somewhere? I can't find any documentation for it.

Comment (one-data-cookie) [reply extensions/chatgpt-quick-actions/CHANGELOG.md]: @JonathanAquino Me neither but it was the same [here](https://github.com/raycast/extensions/pull/13789/commits/2566c32c5b4d15647769cea994aab50daab42f87) in my recent contribution.

File: extensions/chatgpt-quick-actions/package.json (hunk 6 above)
Comment (one-data-cookie) [reply extensions/chatgpt-quick-actions/package.json:372]: @JonathanAquino You're right, ofc – I didn't realise it's `package.json`. Ignore me then! 🙈

**Review by one-data-cookie — APPROVED** [review]
Thanks again, @JonathanAquino!

File: extensions/chatgpt-quick-actions/CHANGELOG.md (hunk 7 above)
Comment (pernielsentikaer) [reply extensions/chatgpt-quick-actions/CHANGELOG.md]: I don't think it's documented, will make sure it will be 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

File: extensions/chatgpt-quick-actions/CHANGELOG.md (hunk 7 above)
Comment (JonathanAquino) [reply extensions/chatgpt-quick-actions/CHANGELOG.md]: @pernielsentikaer it's weird - you merged https://github.com/raycast/extensions/pull/8228 on Sept 4 which documents PR_MERGE_DATE, but the change seems to have gotten reverted in https://github.com/raycast/extensions/commit/24f7f25ccadbdfcc4fd42961d4af8622a02fd9a1 on Sept 6.

**Pull request merged**